### DISCUSSION TITLE
ImportError: cannot import name 'FeedHandler'

### DISCUSSION THREAD

**Discussion by skulltrick** [bmoscon/cryptofeed#480] [Q&A] [answered]
Hi everyone! I have just recently found cryptofeed and am beyond excited to bring some ideas I had to life. However, I am finding myself stumped by a stupid import error. I'm not sure if it is due to how I installed the cryptofeed module, or something I am doing terribly wrong. I tried to find a fix on StackOverflow, but wasn't able to find anything that seemed relevant :(

Here's the sample code:

```
from cryptofeed import FeedHandler
from cryptofeed.exchanges import Coinbase

def nbbo_update(symbol, bid, bid_size, ask, ask_size, bid_feed, ask_feed):
    print(
        f'Pair: {symbol} ' +
        f'Bid Price: {bid:.2f} ' +
        f'Bid Size: {bid_size:.6f} ' +
        f'Bid Feed: {bid_feed} ' +
        f'Ask Price: {ask:.2f} ' +
        f'Ask Size: {ask_size:.6f} ' +
        f'Ask Feed: {ask_feed}'
    )

fh = FeedHandler()
fh.add_nbbo([Coinbase], ['BTC-USD'], nbbo_update)
fh.run()
```

and my output:

`Traceback (most recent call last):
  File "/home/cannon/projects/movingavg/cryptofeed.py", line 1 in <module>
    from cryptofeed import FeedHandler
  File "/home/cannon/projects/movingavg/cryptofeed.py", line 1 in <module>
    from cryptofeed import FeedHandler
ImportError: cannot import name 'FeedHandler`

I've tried `from cryptofeed import *` but then I am given a NameError when FeedHandler is referenced. I have no idea why FeedHandler is not being imported? I installed cryptofeed via pip if that helps at all. Sorry for posting something as simple as an import error, but I am so eager to use cryptofeed and didn't know what else to do.

**Comment by bmoscon** [comment, marked as answer]
What version of python are you using? Also, that code is incomplete, you'll need to import those exchanges you are referencing as well. There is a lot of sample/demo code in `examples/`

**Reply by bmoscon** [reply]
oh, i see the issue, you named your file the same name as the package/module, so that wont work. Rename your python file something other than cryptofeed.

**Reply by skulltrick** [reply]
I'm using python 3.6. Oops that was my mistake, I posted the wrong code earlier. I pasted sample code from my OS, thinking it was the code from my virtual machine. Oh wow! Boy do I feel dumb... Thank you so much for the help! Hopefully this interaction can save some other newbie some time as well haha

**Reply by bmoscon** [reply]
you may or may not have issues depending on what you try and do. Cryptofeed requires 3.7+

**Reply by skulltrick** [reply]
Ahh I see, I will go ahead and update everything then. Thanks again!
